rust/gsc/src/main.rs

```diff
@@ -44,13 +337,21 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let state = Arc::new(connection);
 
     // build our application with a single route
-    let app = Router::new().route(
-        "/job",
-        post({
-            let shared_state = state.clone();
-            move |body| add_job(body, shared_state)
-        }),
-    );
+    let app = Router::new()
+        .route(
+            "/job",
+            post({
+                let shared_state = state.clone();
+                move |body| add_job(body, shared_state)
```

Comment:
Is there a  rusty reason this can't be 

`move |body| add_job(body, state.clone())`

Reply:
`state` would be moved in that case and then I wouldn't be able to share the connection with the other routes. If I didn't have the `move` keyword (which would take a reference to `state` instead) then the function wouldn't implement Send/Sync (or would have some bonkers lifetime shenanigans).